Display lobby options in-game on a dedicated tab

dragunoff opened the pull request:
It's a common frustration that lobby options can't be reviewed once a game has started. Displaying them in-game actually turned out to be not that difficult because the existing chrome logic works just fine in-game. So it became more of an UI problem and most of the changes in this PR deal with the UI. Commit messages give more detail on what's going on and the last couple of commits are mostly polishing. 

At first I thought that the lobby options shouldn't be displayed for missions but then I found that the ones that are not relevant can be easily hidden. That's why I took the time to hide various options from missions and mini-games.

![Screenshot_20210623_140428](https://user-images.githubusercontent.com/1355810/123090662-16722f00-d431-11eb-8f50-0bafa3614434.png)
_All options are displayed for co-op missions, skirmish and multiplayer_
![Screenshot_20210623_140700](https://user-images.githubusercontent.com/1355810/123090675-1b36e300-d431-11eb-81b4-9ec3fad4f1f0.png)
_Only relevant options are displayed for missions and mini-games_

Closes #8396
Related #12349

dragunoff committed:
Display lobby options in-game on a dedicated tab

dragunoff committed:
Do not center lobby options vertically in the scroll panel

dragunoff committed:
Hide various lobby options from missions and minigames

dragunoff committed:
Remove special handling for cases when there's only one game info tab

The addition of the lobby options tab ensures that there will always
be at least two tabs ("Objectives" and "Options").

dragunoff commented:
Update: rebased and resolved conflicts.

abcdefg30 commented:
Do we care about the overflow when the debug menu is enabled or is that also fixed in this PR?

dragunoff commented:
> Do we care about the overflow when the debug menu is enabled or is that also fixed in this PR?

This PR changes the tabs to be left-aligned so that 4 tabs fit neatly. It overflows only on co-op missions (5 tabs):
![Screenshot_20210623_140846](https://user-images.githubusercontent.com/1355810/124399562-d199a280-dd24-11eb-9e5b-b4a1e3707a6b.png)

It's a known issue and a limitation of that design :thinking: I can live with it as it's a rare case to have 5 tabs.

pchote commented:
One of these days we should get around to fixing that button strip properly...

The quickest solution would probably be to define templates for 2/3/4/5 buttons and have the UI logic hook them up similarly to how we generate the ingame menu.

It looks like there is enough padding to be able to fit 5 buttons just by decreasing their width (but only when we need to show 5).

pchote commented:
Filed in #19490.

dragunoff commented:
Rebased on #19490

Smittytron commented:
#19490 has been merged.

dragunoff commented:
Rebased.

Mailaender reviewed (dismissed):
Works as promised.

pchote reviewed (commented):
LGTM overall.

* I realise that the "Map Options" header is to be consistent with the Objectives tab, but the header is redundant vs the tab text and IMO it might look better if it were removed.
* The Difficulty options display empty tooltips
* I'm a bit sad to see the vertical centering removed, but its not a big enough deal to make a fuss over

dragunoff committed:
Add description to difficulty lobby option for missions

dragunoff commented:
Update: rebased and addressed comments.

> 
>  I realise that the "Map Options" header is to be consistent with the Objectives tab, but the header is redundant vs the tab text and IMO it might look better if it were removed.
>
Agreed.

> 
>  The Difficulty options display empty tooltips
> 
This only affected the co-op missions in RA. I added descriptions for the tooltips.

>
>  I'm a bit sad to see the vertical centering removed, but its not a big enough deal to make a fuss over
>
It just looks a little odd in-game - as if there were hidden options:
![image](https://user-images.githubusercontent.com/1355810/134819216-6a48c41f-9580-4af2-b5e3-b52795d2a783.png)

dragunoff commented:
Update: Added the difficulty option description to all missions with a difficulty setting.

pchote commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/9b82679dc07a12e784a4d6f475a849ceaefff461
